docs/src/architecture/08_concepts/hermes_runtime_engine/hre_structure.md

```diff
@@ -0,0 +1,160 @@
+# HRE structure
+
+[Hermes Runtime Extension (*HRE*)][*HRE*] - stands as logically separate module (like a library) of the [*Hermes engine*]
+and provides an additional functionality to the Hermes runtime, therefore to [*Hermes application*].
+[WIT] files represent a source of truth of the [*Hermes events*] and *HRE api* definitions for a specific [*HRE*],
+and describe a standardized communication interface between [*Hermes application*]
+and [*Hermes engine's*][*Hermes engine*] runtime itself.
+
+Each [*HRE*] implementation take place inside `hermes/bin/srs/runtime_extensions` directory,
+for both Hermes related and WASI specific.
+
+Here is an example of the [WIT] files for the some [hermes-cron] [*HRE*]:
+
+`world.wit`
+
+```wit
+package hermes:cron;
+
+world all {
+    import api;
+    export event;
+}
+```
+
+`event.wit`
+
+```wit
+interface event {
+    use api.{cron-event-tag, cron-tagged};
+
+    on-cron: func(event: cron-tagged, last: bool) -> bool;
+}
+
+world cron-event {
+    export event;
+}
+```
+
+`api.wit`
+
+```wit
+interface api {    
+    type cron-event-tag = string;
+
+    type cron-sched = string;
+
+    record cron-tagged {
+        when: cron-sched,
+
+        tag: cron-event-tag
+    }
+
+    add: func(entry: cron-tagged, retrigger: bool) -> bool;
+    rm: func(entry: cron-tagged) -> bool;
+    
+    ...
+}
+
+world cron-api {
+    import api;
+}
+```
+
+*Hermes events*:
+
+* `on-cron: func(event: cron-tagged, last: bool) -> bool;`
+  
+*HRE api*:
+
+* `add: func(entry: cron-tagged, retrigger: bool) -> bool;`
+* `rm: func(entry: cron-tagged) -> bool;`
+
+## Host implementation structure
+
+Hermes host runtime implementation based on the [wasmtime] Rust library.
+It auto-generates a code based on the [WIT] files:
+
+```Rust
+use wasmtime::component::bindgen;
+
+bindgen!({
+    world: "hermes",
+    path: "path/to/the/wit/files/dir",
+});
+```
+
+Under the hood it generates a set of different traits, structs, functions etc.
+which are based on the [WIT] files and provides a type safe interface for interaction
+with WASM modules and for implementation host functionality.
+
+All host implementations for a specific [*HRE*]
+are defined inside corresponded `../hermes/cron/host.rs` and looks like this:
```

Comment:
```suggestion
Internally, it generates a diverse set of traits, structs, functions, and more derived from the WIT files. 
This process results in a type-safe interface for interacting with WASM modules and implementing host functionalities.

All host implementations specific to a particular [*HRE*] are defined within the corresponding
`host.rs` files.

For example, here is an example for `../hermes/cron/host.rs `:
```

ps: I'm not sure if I understood the last two lines corectly.

Reply:
All good, it was specifically what I meant

Reply:
There is a difference in what I proposed and what you added.  
You said that ALL host implementation live in `/hermes/cron/host.rs`.